# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/*/libgpg-error/hotfix.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- libgpg-error-1.42/src/gen-lock-obj.sh	2021-03-04 11:05:29.000000000 +0100
+++ libgpg-error-1.41/src/gen-lock-obj.sh	2020-05-29 14:27:14.000000000 +0200
@@ -2,7 +2,7 @@
 #
 # gen-lock-obj.sh - Build tool to construct the lock object.
 #
-# Copyright (C) 2020, 2021  g10 Code GmbH
+# Copyright (C) 2020 g10 Code GmbH
 #
 # This file is part of libgpg-error.
 #
@@ -38,29 +38,6 @@
 #     AWK=gawk ./gen-lock-obj.sh
 #
 
-if test -n `echo -n`; then
-    ECHO_C='\c'
-    ECHO_N=''
-else
-    ECHO_C=''
-    ECHO_N='-n'
-fi
-
-if test "$1" = --disable-threads; then
-    cat <<EOF
-## lock-obj-pub.$host.h - NO LOCK SUPPORT
-## File created by gen-lock-obj.sh - DO NOT EDIT
-## To be included by mkheader into gpg-error.h
-
-/* Dummy object - no locking available.  */
-typedef struct
-{
-  long _vers;
-} gpgrt_lock_t;
-
-#define GPGRT_LOCK_INITIALIZER {-1}
-EOF
-else
 AWK_VERSION_OUTPUT=$($AWK 'BEGIN { print PROCINFO["version"] }')
 if test -n "$AWK_VERSION_OUTPUT"; then
     # It's GNU awk, which supports PROCINFO.
@@ -107,21 +84,20 @@
 #     USE_LONG_DOUBLE_FOR_ALIGNMENT
 #
 
-echo ${ECHO_N} "#define GPGRT_LOCK_INITIALIZER {$LOCK_ABI_VERSION,{{${ECHO_C}"
+echo -n "#define GPGRT_LOCK_INITIALIZER {$LOCK_ABI_VERSION,{{"
 
 i=0
 while test "$i" -lt $ac_mtx_size; do
     if test "$i" -ne 0 -a "$(( $i % 8 ))" -eq 0; then
         echo ' \'
-        echo ${ECHO_N} "                                    ${ECHO_C}"
+        echo -n "                                    "
     fi
-    echo ${ECHO_N} "0${ECHO_C}"
+    echo -n '0'
     if test "$i" -lt $(($ac_mtx_size - 1)); then
-        echo ${ECHO_N} ",${ECHO_C}"
+        echo -n ','
     fi
     i=$(( i + 1 ))
 done
-fi
 
 cat <<'EOF'
 }}}
